lua/cmp/init.lua

```diff
@@ -110,6 +110,65 @@ cmp.confirm = function(option)
   end
 end
 
+---Show status
+cmp.status = function()
+  local kinds = {}
+  kinds.available = {}
+  kinds.unavailable = {}
+  kinds.installed = {}
+  kinds.invalid = {}
+  local names = {}
+  for _, s in pairs(core.sources) do
+    names[s.name] = true
+
+    if config.get_source_config(s.name) then
+      if s:is_available() then
+        table.insert(kinds.available, s:get_debug_name())
+      else
+        table.insert(kinds.unavailable, s:get_debug_name())
+      end
+    else
+      table.insert(kinds.installed, s:get_debug_name())
+    end
+  end
+  for _, s in ipairs(config.get().sources) do
+    if not names[s.name] then
+      table.insert(kinds.invalid, s.name)
+    end
+  end
+
+  if #kinds.available > 0 then
+    vim.api.nvim_echo({ { '# ready\n', 'Special' } }, false, {})
+    for _, name in ipairs(kinds.available) do
+      vim.api.nvim_echo({ { ('- %s\n'):format(name), 'Normal' } }, false, {})
+    end
+    vim.api.nvim_echo({ { '\n', 'Normal' } }, false, {})
+  end
+
+  if #kinds.unavailable > 0 then
+    vim.api.nvim_echo({ { '# can\'t work\n', 'Comment' } }, false, {})
```

Comment:
"can't work" or "does not work"?

Reply:
Thank you I've fixed it as `unavailable`